src/lib/comms/sol-oic-server.c

```diff
@@ -95,7 +95,13 @@ static sol_coap_responsecode_t
 _sol_oic_server_d(const struct sol_network_link_addr *cliaddr, const void *data,
     const struct sol_oic_map_reader *input, struct sol_oic_map_writer *output)
 {
+    SOL_BUFFER_DECLARE_STATIC(dev_id, 37);
     bool ret;
+    int r;
+
+    r = sol_util_uuid_string_from_bytes(true, true,
+        sol_platform_get_machine_id_as_bytes(), &dev_id);
+    SOL_INT_CHECK_GOTO(r, < 0, error);
```

Comment:
I don't get from this patch why this would not be filled elsewhere anymore... Can we do it at one place only?